### PR TITLE
[flang] Carry over alignment computed by frontend for COMMON

### PR DESCRIPTION
**Pull request by clementval**
The frontend computes the necessary alignment for COMMON blocks but this information is never carried over to the code generation and can lead to segfault for COMMON block that requires a non default alignment. 

This patch add an optional attribute on fir.global and carries over the information.

### CODE REVIEW

File: flang/lib/Lower/ConvertVariable.cpp [hunk 1]
```diff
@@ -1307,7 +1307,10 @@ declareCommonBlock(Fortran::lower::AbstractConverter &converter,
     auto vecTy = mlir::VectorType::get(sz, i8Ty);
     mlir::Attribute zero = builder.getIntegerAttr(i8Ty, 0);
     auto init = mlir::DenseElementsAttr::get(vecTy, llvm::ArrayRef(zero));
-    builder.createGlobal(loc, commonTy, commonName, linkage, init);
+    global = builder.createGlobal(loc, commonTy, commonName, linkage, init);
+    if (const auto *details =
+            common.detailsIf<Fortran::semantics::CommonBlockDetails>())
```
Comment (kiranchandramohan) [inline flang/lib/Lower/ConvertVariable.cpp:1312]: Nit: Is this always true? Should it be an assert?

**Review by kiranchandramohan — APPROVED** [review]
LG.

**Review by jeanPerier — APPROVED** [review]
Thanks!

File: flang/lib/Lower/ConvertVariable.cpp (hunk 1 above)
Comment (clementval) [reply flang/lib/Lower/ConvertVariable.cpp]: It should. I updated the code to always get the details and added an assertion.

**Pull request merged**